snmp/datadog_checks/snmp/snmp.py

```diff
@@ -373,16 +375,21 @@ def _check_with_config(self, config):
             self.warning(error)
         finally:
             # Report service checks
-            sc_tags = ['snmp_device:{}'.format(instance['ip_address'])]
-            sc_tags.extend(instance.get('tags', []))
             status = self.OK
             if error:
                 status = self.CRITICAL
                 if results:
                     status = self.WARNING
-            self.service_check(self.SC_STATUS, status, tags=sc_tags, message=error)
+            self.service_check(self.SC_STATUS, status, tags=tags, message=error)
         return error
 
+    def extract_metric_tags(self, metric_tags, results):
+        extracted_tags = []
+        for tag in metric_tags:
+            result = list(results[tag.symbol].items())[0]
+            extracted_tags.append('{}:{}'.format(tag.name, result[1]))
```

Comment:
Can we use intermediate variables to clarify what these `[0]` and `[1]` index lookups correspond to? I assume `result[1]` is the tag value, but really not sure what `[0]` on L389 refers to.

Reply:
I tried something.